compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -449,16 +449,18 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         }
 
         let result = ensure_sufficient_stack(|| {
-            match obligation.predicate.skip_binders() {
+            let bound_predicate =
+                obligation.predicate.bound_atom_with_opt_escaping(self.infcx().tcx);
```

Comment:
why do you use `...with_opt_escaping` here? We previously used `skip_binders` which failed on unbound vars

Reply:
Whoops